Snabble/UI/Payment/Checks/BaseCheckViewController.swift

```diff
@@ -0,0 +1,309 @@
+            arrow.centerYAnchor.constraint(equalTo: arrowWrapper.centerYAnchor)
+        ])
+
+        codeWrapper.addSubview(code)
+        code.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            code.topAnchor.constraint(equalTo: codeWrapper.topAnchor, constant: 16),
+            code.bottomAnchor.constraint(equalTo: codeWrapper.bottomAnchor, constant: -16),
+            code.centerXAnchor.constraint(equalTo: codeWrapper.centerXAnchor)
+        ])
+
+        idWrapper.addSubview(id)
+        id.translatesAutoresizingMaskIntoConstraints = false
+        id.font = .systemFont(ofSize: 13)
+        NSLayoutConstraint.activate([
+            idWrapper.heightAnchor.constraint(equalToConstant: 21),
+            id.centerXAnchor.constraint(equalTo: idWrapper.centerXAnchor),
+            id.centerYAnchor.constraint(equalTo: idWrapper.centerYAnchor)
+        ])
+
+        arrangeLayout()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        self.navigationItem.hidesBackButton = true
+        self.cancelButton.alpha = 0
+        self.cancelButton.isUserInteractionEnabled = false
+        self.cancelButton.addTarget(self, action: #selector(self.cancelPayment), for: .touchUpInside)
+
+        self.initialBrightness = UIScreen.main.brightness
+        Timer.scheduledTimer(withTimeInterval: 2, repeats: false) { _ in
+            UIView.animate(withDuration: 0.2) {
+                self.cancelButton.alpha = 1
+            }
+            self.cancelButton.isUserInteractionEnabled = true
+        }
+
+        UIApplication.shared.isIdleTimerDisabled = true
+
+        if self.initialBrightness < 0.5 {
+            UIScreen.main.brightness = 0.5
+            self.delegate?.track(.brightnessIncreased)
+        }
+
+        self.setIcon()
+
+        let codeContent = codeContent()
+        self.code.image = renderCode(codeContent)
+
+        let onlineMessageKey = "Snabble.Payment.Online.message"
+        let onlineMessage = SnabbleAPI.l10n(onlineMessageKey)
+        self.text.text = onlineMessage
+        // hide if there is no text/translation
+        self.textWrapper.isHidden = onlineMessage == onlineMessageKey.uppercased()
+
+        self.id.text = String(checkoutProcess.id.suffix(4))
+
+        self.startTimer()
+    }
+
+    override public func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        UIScreen.main.brightness = self.initialBrightness
+        UIApplication.shared.isIdleTimerDisabled = false
+    }
+
+    // MARK: - override points
+    func arrangeLayout() {
+        fatalError("clients must override")
+    }
+
+    func renderCode(_ content: String) -> UIImage? {
+        fatalError("clients must override")
+    }
+
+    func checkContinuation(for process: CheckoutProcess) -> CheckResult {
+        fatalError("clients must override")
+    }
```

Comment:
Durch den noch folgenden Vorschlag würden die Methoden entfallen und die Superklasse müsste auch keinerlei Wissen über die genaue visuelle Darstellungen ihres Content haben. 

Ist der Superklasse nur wichtig das die Methoden  `renderCode(_:)` und `checkContinuation(for:)` implementiert werden? Falls dem so ist, warum bauen wir noch die zusätzliche Abhängigkeit zur visuellen Darstellung auf?

- Durch die Verwendung von Containment und einem Protokoll würden Abhängigkeiten aufgelöst werden.
- Es gibt keine `arrangeLayout` Methode die jeder Entwickler nachvollziehen muss.
- Das Protokoll definiert eindeutig was die Kinder bereitstellen müssen.